task/push-build-file/0.1/push-build-file.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "docker"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "image-build, appstudio"
+  name: push-build-file
+spec:
+  description: |-
+    Discover build file from source code and push it to registry as an OCI artifact. The build file could be named Dockerfile or Containerfile, which can be specified via parameter.
+  params:
+  - name: IMAGE
+    description: The built binary image. The build file is pushed to the same image repository alongside.
+    type: string
+  - name: IMAGE_DIGEST
+    description: The built binary image digest, which is used to construct the tag of build file image.
+    type: string
+  - name: BUILD_FILE
+    description: Path to the build file.
+    type: string
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: Path to the directory to use as context.
+    type: string
+    default: .
+  - name: TAG_SUFFIX
+    description: Suffix of the build file image tag.
+    type: string
+    default: .dockerfile
+  - name: ARTIFACT_TYPE
+    description: Artifact type of the build file image.
+    type: string
+    default: application/vnd.konflux.dockerfile
+  results:
+  - name: IMAGE_URL
+    description: Digest-pinned image URL.
```

Comment:
Let's maybe use `IMAGE_REF`? Build tasks use `IMAGE_URL` as the digest-less repo url

Reply:
Maybe. But, I'm thinking if we should use consistent result name, since we haven't had separated concept for digest-pinned and digest-less image name yet.

Reply:
There's been a proposal to add an IMAGE_REF result to the buildah task (a digest-pinned url): https://github.com/konflux-ci/build-definitions/pull/1111

I think `IMAGE_URL` and `IMAGE_DIGEST` also have special meaning to Tekton Chains

Reply:
Oh, haven't noticed that PR. Give me a sec, let me update this PR.

Reply:
PTAL.